Pyrado/pyrado/algorithms/meta/sprl.py

```diff
@@ -0,0 +1,573 @@
+        if not (len(stacked.shape) == 1):
+            raise pyrado.ValueErr(msg="Stacked has invalid shape! Must be 1-dimensional.")
+
+        expected_dim_multiplier = 0
+        if self._mean_is_parameter:
+            expected_dim_multiplier += 1
+        if self._cov_is_parameter:
+            expected_dim_multiplier += 1
+        if not (stacked.shape[0] == expected_dim_multiplier * self.dim):
+            raise pyrado.ValueErr(msg=f"Stacked has invalid size! Must be {expected_dim_multiplier}*dim.")
+
+        if self._mean_is_parameter and self._cov_is_parameter:
+            mean = stacked[: self.dim]
+            cov_chol_flat = stacked[self.dim :]
+        elif self._mean_is_parameter and not self._cov_is_parameter:
+            mean = stacked[: self.dim]
+            cov_chol_flat = self.cov_chol_flat
+        elif not self._mean_is_parameter and self._cov_is_parameter:
+            mean = self.mean
+            cov_chol_flat = stacked
+        else:
+            mean = self.mean
+            cov_chol_flat = self.cov_chol_flat
+
+        if type(mean) == np.ndarray:
+            mean = to.tensor(mean).double()
+
+        if type(cov_chol_flat) == np.ndarray:
+            cov_chol_flat = to.tensor(cov_chol_flat).double()
+
+        return ParameterAgnosticMultivariateNormalWrapper(
+            mean=mean,
+            cov_chol_flat=cov_chol_flat,
+            mean_is_parameter=self._mean_is_parameter,
+            cov_is_parameter=self._cov_is_parameter,
+        )
+
+    def parameters(self) -> List[to.Tensor]:
+        """ Gets the list of parameters according to the values passed to the constructor. """
+        params = []
+        if self._mean_is_parameter:
+            params.append(self.mean)
+        if self._cov_is_parameter:
+            params.append(self.cov_chol_flat)
+        return params
+
+    def get_stacked(
+        self, return_mean_cov_indices: bool = False
+    ) -> Union[np.ndarray, Tuple[np.ndarray, Optional[List[int]], Optional[List[int]]]]:
+        """
+        Like `MultivariateNormalWrapper.get_stacked(self)`, returns a numpy array with the mean and standard deviations
+        stacked on top of each other if the respective value is a parameter. Additionally, if `return_mean_cov_indices`
+        is `True`, also the indices of the mean/cov values are returned which can be used for addressing them.
+
+        :param return_mean_cov_indices: if `True`, additionally to just the parameters also the indices are returned
+        :return: if `return_mean_cov_indices` is `True`, a tuple `(stacked, mean_indices, cov_indices)`, where the
+                 latter two might be `None` if the respective value is not a parameter; if `return_mean_cov_indices`
+                 is `False`, just the stacked values; the stacked values can have shape `(0,)`, `(k,)`, or `(2 * k,)`,
+                 depending on if none, only mean/cov, or both are parameters, respectively
+        """
+        parameters = self.parameters()
```

Comment:
see other comments above

Reply:
Done in bcdc32aa4a8d02ea06ca04b85098b97bb60c1447